### DISCUSSION TITLE
Vaillant atmoTec exclusive

### DISCUSSION THREAD

**Discussion by joehaens** [john30/ebusd#1212] [Q&A]
Hi all,
does anyone have experience reading and writing values using a Vaillant atmoTec exlusive gas heater? I attached the output of ebusctl below. After setting `EBUSD_MQTTVAR: "filter-direction=r|u|^w"` in the docker-compose.yml, I do not get any writeable variable for the boiler (bai) in HA.

`version: ebusd 23.3.23.3
update check: OK
signal: acquired
symbol rate: 22
max symbol rate: 62
min arbitration micros: 4
max arbitration micros: 19
min symbol latency: 9
max symbol latency: 30
scan: finished
reconnects: 0
masters: 3
messages: 226
conditional: 3
poll: 71
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0703;HW=7401", loaded "vaillant/bai.308523.inc" ([HW=7401]), "vaillant/08.bai.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=37000;SW=0130;HW=6002"
address 31: master #8, ebusd
address 36: slave #8, ebusd
`

My final intention is to control the flow temperature of the heater, which currently is returned in Status02 as `"2": {"name": "temp1", "value": 40.0},`:
     "0": {"name": "hwcmode", "value": "auto"},
     "1": {"name": "temp0", "value": 60},
     "2": {"name": "temp1", "value": 40.0},
     "3": {"name": "temp0", "value": 70},
     "4": {"name": "temp1", "value": 42.0}}

My question is: 
- Can I somehow check for writeable parameters in a different way
- The parameters published to ebusd topic are different from the ones published to home-assistant, is that by intention?
- For my understanding the loaded bai-file is the configuration, which in this specific case is the ecoTec. I dont know if there are any differences between ecoTec and atmoTec on the ebus communication

**Comment by JonesGW-MGD** [comment]
Hi there,  I may not be able to provide a full response but:

(in reverse order)

The file(s) loaded depends on walking through 08.bai.csv (which is what the auto config does - consider this file like a 'manifest') and determining whether - by product ID or HW byte codes, there is a more precise match of config file.  Often, as in your case and mine, it will load bai.308523.inc because it reaches the end of this manifest file without a match - which we can consider is the 'default Vaillant' gas boiler config setup.

It's almost impossible to work out Vaillant's product naming - it differs by country and a plethora of other factors - who knows!  But atmotec / turbotec / ecotec may all be very similar.

But essentially, if your setup loads bai.308523.csv, it means there is no closer match to your Vaillant configuration in the 'official' ebusd repository.  (You can always clone these files locally and then refine them based on other peoples input/experimentation - but that is another 'ball of wax').

Not sure I understand your second comment - would you care to give an example of your set-up?

Recently I wrote up my experiences/learning curve to writing values - such as 'HeatCurve' - here https://github.com/john30/ebusd/discussions/1177.  Maybe it might help you determine whether there is anything you can write?  (I have some examples of using HA to test-write some values but you can probably short-cut the process to use ebustctl read / then write to test in terminal first?)

Changing flow temp can be problematic - depending on your Vaillant config, because the obvious things to change might be read-only because of things like weather compensation etc.  

There is a 'hack' to this which is to define a custom 'SetModeOverride' message and then write to this - this is documented on the discussion forum so have a look - Again, I referred another user to this in this post: https://github.com/john30/ebusd/discussions/518#discussioncomment-8482931 and suggested this might be a good place to start: https://github.com/john30/ebusd-configuration/issues/242

Be aware that if you attempt to use this to override the other controls on your system (controller, boiler default controls), you may need to issue your command periodically (i.e. every 5-10 mins) otherwise it will revert to default automatically!

hope this helps a little, good luck
